doc/whats_new/v1.0.rst

```diff
@@ -257,6 +257,12 @@ Changelog
   :class:`ensemble.StackingClassifier` and :class:`ensemble.StackingRegressor`.
   :pr:`19564` by `Thomas Fan`_.
 
+- |Fix| Fixed the range of the argument max_samples to be (0.0, 1.0]
+  in :class:`ensemble.RandomForestClassifier`,
+  :class:`ensemble.RandomForestRegressor` and all other
+  random forest based estimators.
+  :pr:`20159` by :user:`murata-yu`.
```

Comment:
I think it would be good to describe what 1.0 means:

```suggestion
  :class:`ensemble.RandomForestRegressor`, where `max_samples=1.0` is
  interpreted as using all `n_samples` for bootstrapping. :pr:`20159` by
  :user:`murata-yu`.
```

We can remove the "all other random forest based estimators" , because only `RandomForestClassifier` and `RandomForestRegressor` allows the user to change `max_samples`.